test/e2e/smoke/with-items-with-artifact-repo-ref.yaml

```
@@ -0,0 +1,43 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: success-optional-artifact-with-items-
+spec:
+  entrypoint: artifact-example
+  templates:
+    - name: artifact-example
+      steps:
+        - - name: generate-artifact
+            template: whalesay
+        - - name: consume-artifact
+            template: print-message
+            arguments:
+              artifacts:
+                - name: message
+                  optional: true
+                  from: "{{steps.generate-artifact.outputs.artifacts.message}}"
+            withItems:
+              - hello world
+
+    - name: whalesay
+      container:
+        image: argoproj/argosay:v1
+        command: [ sh, -c ]
+        args: [ "echo 'hello world' > /tmp/hello_world;" ]
+      outputs:
+        artifacts:
+          - name: message
+            optional: true
+            path: /tmp/hello_world
+
+
+    - name: print-message
+      inputs:
+        artifacts:
+          - name: message
+            optional: true
+            path: /tmp/message
+      container:
+        image: argoproj/argosay:v1
+        command: [ sh, -c ]
+        args: [ "test -e /tmp/message && ls /tmp/message || echo 0" ]
```

I know we came up with this repro workflow, so this is an attempt of me fixing our past mistakes 😅 Shouldn't it be like this?

```suggestion
        args: [ "test -e /tmp/message && ls /tmp/message || echo 0" ]
```